controllers/cloud.redhat.com/namespacepool_controller.go

```diff
@@ -51,21 +52,25 @@ type NamespacePoolReconciler struct {
 
 func (r *NamespacePoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	pool := crd.NamespacePool{}
+
 	if err := r.Client.Get(ctx, req.NamespacedName, &pool); err != nil {
-		r.Log.Error(err, "Error retrieving namespace pool")
-		return ctrl.Result{}, err
+		r.Log.Error(err, fmt.Sprintf("cannot retrieve [%s] pool resource", pool.Name))
+		return ctrl.Result{Requeue: true}, err
 	}
 
 	errNamespaceList, err := r.getPoolStatus(ctx, &pool)
 	if err != nil {
-		r.Log.Error(err, "Unable to get status of owned namespaces")
+		r.Log.Error(err, "unable to get status of owned namespaces")
 		return ctrl.Result{Requeue: true}, err
 	}
 
-	err = r.handleErrorNamespaces(ctx, errNamespaceList)
-	if err != nil {
-		r.Log.Error(err, "Unable to delete object.")
-		return ctrl.Result{Requeue: true}, err
+	if len(errNamespaceList) > 0 {
```

Comment:
Flip and return

Reply:
Will probably talk with you on this one next week.